docs/development/linux-bridge-primary-nic-scenario.md

```diff
@@ -0,0 +1,787 @@
+    next-hop-interface: eth1                                                    
+    table-id: 254                                                               
+                                                                                
+interfaces:                                                                     
+  - name: eth1                                                                  
+    type: ethernet                                                              
+    state: up                                                                   
+    ipv4:                                                                       
+      address:                                                                  
+      - ip: 10.244.0.1                                                          
+        prefix-length: 24                                                       
+`                                                                               
+    currentState := state.State{}                                               
+                                                                                
+    err := yaml.Unmarshal([]byte(currentStateYAML), currentState)               
+    if err != nil {                                                             
+        panic(err)                                                              
+    }                                                                           
+                                                                                
+    capture := map[string]string{                                               
+        "default-gw":               `routes.running.destination=="0.0.0.0/0"`,  
+        "base-iface-routes":        `routes.running.next-hop-interface==capture.default-gw.routes.running.0.next-hop-interface`,                                                     
+        "base-iface":               `interfaces.name==capture.default-gw.routes.running.0.next-hop-interface`,
+        "bridge-routes":            `capture.base-iface-routes | routes.running.next-hop-interface="br1"`,
+        "delete-base-iface-routes": `capture.base-iface-route | routes.running.state="absent"`,
+        "bridge-routes-takeover":   `capture.delete-base-iface-routes.routes.running + capture.bridge-routes.routes.running`,
+    }                                                                           
+                                                                                
+    commandByName := map[string]ast.Command{}                                   
+    for ek, ev := range capture {                                               
+        source := source.New(ev)                                                
+        parser := parser.New(*source, lexer.NewLexer(source))                   
+        command, err := parser.Parse()                                          
```

Comment:
`command`is the AST?

Reply:
The main type is Command, AST is just the name of the package so the AST is a *ast.Command that's ok.